Issue with CwiseBinaryOp on 14.04

#45 raised the following error on 14.04, with gcc:

```
[ 16%] Building CXX object src/CMakeFiles/hpp-fcl.dir/shape/geometric_shapes.cpp.o
cd /home/travis/build/nim65s/hpp-fcl/build/src && /usr/bin/g++  -DFCL_HAVE_OCTOMAP -DFCL_USE_ASSIMP_UNIFIED_HEADER_NAMES -DOCTOMAP_MAJOR_VERSION=1 -DOCTOMAP_MINOR_VERSION=8 -DOCTOMAP_PATCH_VERSION=0 -Dhpp_fcl_EXPORTS -I/home/travis/build/nim65s/hpp-fcl/build -I/home/travis/build/nim65s/hpp-fcl/build/include -I/home/travis/build/nim65s/hpp-fcl/include -I/usr/include/eigen3 -isystem /usr/local/include -isystem /usr/include/assimp  -pedantic -Wno-long-long -Wall -Wextra -Wcast-align -Wcast-qual -Wformat -Wwrite-strings -Wconversion -w -g -fPIC   -o CMakeFiles/hpp-fcl.dir/shape/geometric_shapes.cpp.o -c /home/travis/build/nim65s/hpp-fcl/src/shape/geometric_shapes.cpp
In file included from /usr/include/eigen3/Eigen/src/Core/MatrixBase.h:139:0,
                 from /usr/include/eigen3/Eigen/Core:279,
                 from /home/travis/build/nim65s/hpp-fcl/include/hpp/fcl/math/vec_3f.h:43,
                 from /home/travis/build/nim65s/hpp-fcl/include/hpp/fcl/BV/OBB.h:42,
                 from /home/travis/build/nim65s/hpp-fcl/src/BV/OBB.cpp:38:
/usr/include/eigen3/Eigen/src/Core/../plugins/MatrixCwiseBinaryOps.h: In instantiation of ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_max_op<typename Eigen::internal::traits<T>::Scalar>, const Derived, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<typename Eigen::internal::traits<T>::Scalar>, Derived> > Eigen::MatrixBase<Derived>::cwiseMax(const Scalar&) const [with Derived = Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> >; typename Eigen::internal::traits<T>::Scalar = double; Eigen::MatrixBase<Derived>::Scalar = double]’:
/home/travis/build/nim65s/hpp-fcl/src/BV/OBB.cpp:317:60:   required from here
/usr/include/eigen3/Eigen/src/Core/../plugins/MatrixCwiseBinaryOps.h:110:72: error: could not convert ‘Eigen::MatrixBase<Derived>::cwiseMax(const Eigen::MatrixBase<OtherDerived>&) const [with OtherDerived = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; Derived = Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> >; typename Eigen::internal::traits<T>::Scalar = double]((*(const Eigen::MatrixBase<Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> > >*)(& Eigen::DenseBase<Derived>::Constant(Eigen::DenseBase<Derived>::Index, Eigen::DenseBase<Derived>::Index, const Scalar&) [with Derived = Eigen::Matrix<double, 3, 1>; Eigen::DenseBase<Derived>::ConstantReturnType = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> >; typename Eigen::internal::traits<T>::Scalar = double; Eigen::DenseBase<Derived>::Index = long int; Eigen::DenseBase<Derived>::Scalar = double](((const Eigen::EigenBase<Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> > >*)this)->Eigen::EigenBase<Derived>::cols<Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> > >(), (* & other)))))’ from ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_max_op<double>, const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> >, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Matrix<double, 3, 1> > >’ to ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_max_op<double>, const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> >, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const Eigen::Matrix<double, 3, 1>, const Eigen::Matrix<double, 3, 1> > > >’
   return cwiseMax(Derived::PlainObject::Constant(rows(), cols(), other));
```

and clang:
```
[ 16%] Building CXX object src/CMakeFiles/hpp-fcl.dir/shape/geometric_shapes.cpp.o
cd /home/travis/build/nim65s/hpp-fcl/build/src && /usr/local/clang-5.0.0/bin/clang++  -DFCL_HAVE_OCTOMAP -DFCL_USE_ASSIMP_UNIFIED_HEADER_NAMES -DOCTOMAP_MAJOR_VERSION=1 -DOCTOMAP_MINOR_VERSION=8 -DOCTOMAP_PATCH_VERSION=0 -Dhpp_fcl_EXPORTS -I/home/travis/build/nim65s/hpp-fcl/build -I/home/travis/build/nim65s/hpp-fcl/build/include -I/home/travis/build/nim65s/hpp-fcl/include -I/usr/include/eigen3 -isystem /usr/local/include -isystem /usr/include/assimp  -pedantic -Wno-long-long -Wall -Wextra -Wcast-align -Wcast-qual -Wformat -Wwrite-strings -Wconversion -w -g -fPIC   -o CMakeFiles/hpp-fcl.dir/shape/geometric_shapes.cpp.o -c /home/travis/build/nim65s/hpp-fcl/src/shape/geometric_shapes.cpp
In file included from /home/travis/build/nim65s/hpp-fcl/src/BV/OBB.cpp:38:
In file included from /home/travis/build/nim65s/hpp-fcl/include/hpp/fcl/BV/OBB.h:42:
In file included from /home/travis/build/nim65s/hpp-fcl/include/hpp/fcl/math/vec_3f.h:43:
In file included from /usr/include/eigen3/Eigen/Core:279:
In file included from /usr/include/eigen3/Eigen/src/Core/MatrixBase.h:139:
/usr/include/eigen3/Eigen/src/Core/../plugins/MatrixCwiseBinaryOps.h:110:10: error: 
      no viable conversion from returned value of type 'const CwiseBinaryOp<[2 *
      ...], const CwiseNullaryOp<[...], Eigen::Matrix<double, 3, 1, 0, 3,
      1>>>' to function return type 'const CwiseBinaryOp<[2 * ...], const
      CwiseNullaryOp<[...],
      Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const
      Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3, 1, 0,
      3, 1> >>>'
  return cwiseMax(Derived::PlainObject::Constant(rows(), cols(), other));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/travis/build/nim65s/hpp-fcl/src/BV/OBB.cpp:317:49: note: in instantiation
      of member function
      'Eigen::MatrixBase<Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>,
      const Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3,
      1, 0, 3, 1> > >::cwiseMax' requested here
  squaredLowerBoundDistance = (AABB_corner - a).cwiseMax (0).squaredNorm ();
                                                ^
/usr/include/eigen3/Eigen/src/Core/util/ForwardDeclarations.h:89:65: note: 
      candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'const CwiseBinaryOp<internal::scalar_max_op<Scalar>,
      const Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>,
      const Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3,
      1, 0, 3, 1> >, const
      Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>,
      Eigen::Matrix<double, 3, 1, 0, 3, 1> > >' (aka 'const
      CwiseBinaryOp<scalar_max_op<double>, const
      Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const
      Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3, 1, 0,
      3, 1> >, const
      Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>,
      Eigen::Matrix<double, 3, 1, 0, 3, 1> > >') to 'const
      Eigen::CwiseBinaryOp<Eigen::internal::scalar_max_op<double>, const
      Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const
      Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3, 1, 0,
      3, 1> >, const
      Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>,
      Eigen::CwiseBinaryOp<Eigen::internal::scalar_difference_op<double>, const
      Eigen::Matrix<double, 3, 1, 0, 3, 1>, const Eigen::Matrix<double, 3, 1, 0,
      3, 1> > > > &' for 1st argument
template<typename BinaryOp,  typename Lhs, typename Rhs>  class CwiseBinaryOp;
                                                                ^
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,8 +10,6 @@ matrix:
       compiler: gcc
     - dist: xenial
       compiler: gcc
-    - dist: trusty
-      compiler: clang
     - dist: xenial
       compiler: clang
     - os: osx

--- a/src/BV/OBB.cpp
+++ b/src/BV/OBB.cpp
@@ -314,13 +314,18 @@ bool obbDisjointAndLowerBoundDistance (const Matrix3f& B, const Vec3f& T,
 
   // Corner of b axis aligned bounding box the closest to the origin
   Vec3f AABB_corner (T.cwiseAbs () - Bf * b);
-  squaredLowerBoundDistance = (AABB_corner - a).cwiseMax (0).squaredNorm ();
+  Vec3f diff3 (AABB_corner - a);
+  diff3 = diff3.cwiseMax (0);
+  //for (Vec3f::Index i=0; i<3; ++i) diff3 [i] = std::max (0, diff3 [i]);
+  squaredLowerBoundDistance = diff3.squaredNorm ();
   if (squaredLowerBoundDistance > breakDistance2)
     return true;
 
   AABB_corner = (B.transpose () * T).cwiseAbs ()  - Bf.transpose () * a;
-  // | B^T T| - b - Bf^T a
-  squaredLowerBoundDistance = (AABB_corner - b).cwiseMax (0).squaredNorm ();
+  // diff3 = | B^T T| - b - Bf^T a
+  diff3 = AABB_corner - b;
+  diff3 = diff3.cwiseMax (0);
+  squaredLowerBoundDistance = diff3.squaredNorm ();
 
   if (squaredLowerBoundDistance > breakDistance2)
     return true;

